modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/value/ValueProviderUtils.java

```diff
@@ -118,17 +141,22 @@ static ValueBuilder cloneAndEnrichValue(Value value, Map<Integer, String> partOr
    *
    * @param valueProviderClazz a class that implements the {@link ValueProvider} interface.
    * @return The id of the value provider
+   *
    * @since 4.4.0
    */
-  public static String getValueProviderId(Class<? extends ValueProvider> valueProviderClazz) {
-    ValueProvider valueProvider;
+  public static String getValueProviderId(Class valueProviderClazz) {
+    Object valueProviderObject;
     try {
-      valueProvider = instantiateClass(valueProviderClazz);
+      valueProviderObject = instantiateClass(valueProviderClazz);
     } catch (Exception e) {
       throw new IllegalStateException(format("There was an error creating an instance of %s to retrieve the Id of the provider",
                                              valueProviderClazz.getName()),
                                       e);
     }
-    return valueProvider.getId();
+
+    return valueProviderObject instanceof org.mule.runtime.extension.api.values.ValueProvider
+        ? ((org.mule.runtime.extension.api.values.ValueProvider) valueProviderObject).getId()
+        : ((ValueProvider) valueProviderObject).getId();
```

Comment:
I think we should also check that it is a class that implements ValueProvider instead of assuming it because it is not an sdk VP.

Reply:
The entire code needs to be writteng in terms of the SDK API. We shouldn't have these type of checks in any where else other than the ValuePRoviderFactory.